cmd/applyconfig/applyconfig.go

```diff
@@ -54,23 +65,46 @@ func (n *nullableStringFlag) Set(val string) error {
 }
 
 func gatherOptions() *options {
-	opt := &options{user: &nullableStringFlag{}}
-	flag.BoolVar(&opt.confirm, "confirm", false, "Set to true to make applyconfig commit the config to the cluster")
+	// nonempy dryRun is a safe default (empty means not a dry run)
+	// TODO(muller): client dry run is default until we pass server one.
+	opt := &options{user: &nullableStringFlag{}, dryRun: dryClient}
+
+	var confirm bool
+	flag.BoolVar(&confirm, "confirm", false, "Set to true to make applyconfig commit the config to the cluster")
 	flag.Var(opt.user, "as", "Username to impersonate while applying the config")
 	flag.Var(&opt.directories, "config-dir", "Directory with config to apply. Can be repeated multiple times.")
 	flag.StringVar(&opt.context, "context", "", "Context name to use while applying the config")
 	flag.StringVar(&opt.kubeConfig, "kubeconfig", "", "Path to the kubeconfig file to apply the config")
+
+	var dryMethod string
+	dryRunMethods := strings.Join(validDryRunMethods, ",")
+	flag.StringVar(&dryMethod, "dry-run-method", string(opt.dryRun), fmt.Sprintf("Method to use when running when --confirm is not set to true (valid values: %s)", dryRunMethods))
```

Comment:
This seems overly complicated to me. Why not simply add a boolean for one of the two dry runs, then use dry-run when confirm=calse and switch the method based on the bool?